diff --git a/arch/riscv/dts/sun20i-d1-lichee-rv-dock.dts b/arch/riscv/dts/sun20i-d1-lichee-rv-dock.dts
index bcdd8088c7..db169006f6 100644
--- a/arch/riscv/dts/sun20i-d1-lichee-rv-dock.dts
+++ b/arch/riscv/dts/sun20i-d1-lichee-rv-dock.dts
@@ -13,6 +13,8 @@
 
 	aliases {
 		ethernet1 = &rtl8723ds;
+		i2c2 = &i2c2;
+		spi1 = &spi1;
 	};
 
 	dmic_codec: dmic-codec {
@@ -170,3 +172,21 @@
 &usbphy {
 	usb1_vbus-supply = <&reg_vcc>;
 };
+
+&i2c2 {
+	pinctrl-0 = <&i2c2_pb0_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&spi1 {
+	pinctrl-0 = <&spi1_pd_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+	spidev@0 {
+		compatible = "linux,spidev", "rohm,dh2228fv" ;
+		spi-max-frequency = <100000000>;
+		reg = <0>;
+		status = "okay";
+	};
+};
diff --git a/arch/riscv/dts/sun20i-d1-mangopi-mq-pro.dts b/arch/riscv/dts/sun20i-d1-mangopi-mq-pro.dts
index a7149baf18..21c626f14b 100644
--- a/arch/riscv/dts/sun20i-d1-mangopi-mq-pro.dts
+++ b/arch/riscv/dts/sun20i-d1-mangopi-mq-pro.dts
@@ -17,6 +17,8 @@
 		ethernet0 = &rtl8723ds;
 		mmc0 = &mmc0;
 		serial0 = &uart0;
+		i2c0 = &i2c0;
+		spi1 = &spi1;
 	};
 
 	chosen {
@@ -167,3 +169,21 @@
 	usb0_vbus-supply = <&reg_vcc>;
 	status = "okay";
 };
+
+&i2c0 {
+	pinctrl-0 = <&i2c0_pg12_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&spi1 {
+	pinctrl-0 = <&spi1_pd_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+	spidev@0 {
+		compatible = "linux,spidev", "rohm,dh2228fv" ;
+		spi-max-frequency = <100000000>;
+		reg = <0>;
+		status = "okay";
+	};
+};
diff --git a/arch/riscv/dts/sun20i-d1.dtsi b/arch/riscv/dts/sun20i-d1.dtsi
index e218c9d681..74288262a2 100644
--- a/arch/riscv/dts/sun20i-d1.dtsi
+++ b/arch/riscv/dts/sun20i-d1.dtsi
@@ -137,6 +137,12 @@
 				function = "i2c0";
 			};
 
+			/omit-if-no-ref/
+			i2c0_pg12_pins: i2c0-pg12-pins {
+				pins = "PG12", "PG13";
+				function = "i2c0";
+			};
+
 			/omit-if-no-ref/
 			i2c2_pb0_pins: i2c2-pb0-pins {
 				pins = "PB0", "PB1";
